### PR TITLE
add dark mode and shimmer UI

### LINKED ISSUE #6: Feat : Dark mode

### PR DESCRIPTION
**Pull request by swarajsahil**
Feat : Dark mode #6 Add dark mode using context Api and replace loading with shimmer UI
Screenshot 2023-05-31 143125
Screenshot 2023-05-31 143155

### CODE REVIEW

**Review by yuvrxj-afk — COMMENTED** [review]
align the dark mode switch to right.

**Comment by swarajsahil** [comment]
@yuvrxj-afk done align the dark mode switch to right.

File: src/pages/Home.jsx
Comment (yuvrxj-afk) [inline src/pages/Home.jsx:1]: Posts aren't showing up !
give it a look.

Comment (swarajsahil) [reply src/pages/Home.jsx:1]: i comment wait solving it

**Review by yuvrxj-afk — COMMENTED** [review]
check up the pile

File: src/pages/Home.jsx
Comment (yuvrxj-afk) [reply src/pages/Home.jsx:1]: alright.

Comment (swarajsahil) [reply src/pages/Home.jsx:1]: Now check showing the card

**Comment by yuvrxj-afk** [comment]
Would you also fix the padding of create post page's default generating image? 
and there wasn't any top border on the home page , just the bottom border navbar would be great!

**Comment by swarajsahil** [comment]
ok i try to solve it any suggestion for create post page default generating image how it look

File: src/pages/Home.jsx [hunk 1]
```diff
@@ -83,8 +96,10 @@ const Home = () => {
 
       <div className="mt-18">
         {loading ? (
-          <div className="flex justify-center items-center mt-4">
-            <Loader />
+          <div className="flex flex-wrap justify-center gap-5 mt-5">
+            <Loader /><Loader /><Loader />
+            <Loader /><Loader /><Loader />
+            <Loader /><Loader /><Loader />
```
Comment (yuvrxj-afk) [inline src/pages/Home.jsx]: doesn't seem a good practice , map it as an array(9).

Comment (swarajsahil) [reply src/pages/Home.jsx]: ok 
and  any suggestion for create post page default generating image how it look

Comment (yuvrxj-afk) [reply src/pages/Home.jsx]: nothing particular , just the image
this

File: src/components/Loader.jsx [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+  //     <path d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z" fill="currentColor" />
+  //     <path d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z" fill="currentFill" />
+  //   </svg>
+  // </div>
+            <div className="animate-pulse w-44 h-48 bg-slate-400 rounded-lg"></div>
```
Comment (yuvrxj-afk) [inline src/components/Loader.jsx]: clear the comments too.

Comment (swarajsahil) [reply src/components/Loader.jsx]: Screenshot 2023-06-01 113436
how it look any suggestion and any design you have

Comment (yuvrxj-afk) [reply src/components/Loader.jsx]: No for now , if you have any pin it up.

**Comment by yuvrxj-afk** [comment]
Re-mentioning the tasks requested :

- Fix the border of navigation bar.
- Clear the comments // (Loader.jsx)
- Frame the generating image in a proper aligned way.
- Also add a Contribute button (github Icon) in the bar which directs to repository.

**Comment by swarajsahil** [comment]
@yuvrxj-afk 
#9 Done all this.
Fix the border of navigation bar.
Clear the comments // (Loader.jsx)
Frame the generating image in a proper aligned way.
Also add a Contribute button (github Icon) in the bar which directs to repository.
Screenshot 2023-06-01 154727

**Comment by yuvrxj-afk** [comment]
fix the padding between the right icons in the navbar , justify between them.

**Comment by swarajsahil** [comment]
Okay

**Comment by swarajsahil** [comment]
#9  I check and  make some changes it look good .

**Comment by yuvrxj-afk** [comment]
> #9 I check and make some changes it look good .

you removed the setLoading , which acts as a loader when the picture gets generated.

**Comment by swarajsahil** [comment]
When the picture is generated I put a new blur picture with a little opacity

**Pull request merged**